Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -107,6 +124,45 @@ class HistoryPanel: SiteTableViewController, HomePanel {
         }
     }
 
+    private func createEmptyStateOverview() -> UIView {
+        let overlayView = UIView(frame: tableView.bounds)
+        overlayView.backgroundColor = UIColor.whiteColor()
+        // Unknown why this does not work with autolayout
+        overlayView.autoresizingMask = UIViewAutoresizing.FlexibleHeight | UIViewAutoresizing.FlexibleWidth
+
+        let containerView = UIView()
+        overlayView.addSubview(containerView)
+
+        let logoImageView = UIImageView(image: UIImage(named: "emptyHistory"))
+        containerView.addSubview(logoImageView)
+        logoImageView.snp_makeConstraints({ (make) -> Void in
+            make.centerX.equalTo(containerView)
+            make.top.equalTo(containerView)
+        })
+
+        let welcomeLabel = UILabel()
+        containerView.addSubview(welcomeLabel)
+        welcomeLabel.text = NSLocalizedString("Pages you have visited recently will show up here.", comment: "See http://mzl.la/1LXbDOL")
+        welcomeLabel.textAlignment = NSTextAlignment.Center
+        welcomeLabel.font = HistoryPanelUX.WelcomeScreenItemFont
+        welcomeLabel.textColor = HistoryPanelUX.WelcomeScreenItemTextColor
+        welcomeLabel.numberOfLines = 2
+        welcomeLabel.adjustsFontSizeToFitWidth = true
+
+        welcomeLabel.snp_makeConstraints({ (make) -> Void in
+            make.centerX.equalTo(containerView)
+            make.top.equalTo(logoImageView.snp_bottom).offset(HistoryPanelUX.WelcomeScreenPadding)
+            make.width.equalTo(HistoryPanelUX.WelcomeScreenItemWidth)
+        })
+
+        containerView.snp_makeConstraints({ (make) -> Void in
+            make.top.equalTo(overlayView.snp_top).offset(20)
+            make.centerX.equalTo(overlayView)
+        })
```

Comment:
Not sure I see the value of having `containerView`. Could you just add the logo and label directlly to `overlayView`?